### PR TITLE
fix(1710): Create join job and store external parentBuilds info [3]

### PR DESCRIPTION
**Pull request by tkyi**
## Context

Currently the external parallel fork join feature does not trigger the join build after both parallel builds complete. This is because the external build is not storing its information in the join build properly.

## Objective

This PR creates the join build immediately after creating the external build so that the parentBuilds field is populated correctly.

## References

Related to https://github.com/screwdriver-cd/screwdriver/issues/1710

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/builds/index.js [hunk 1]
```diff
@@ -753,56 +872,81 @@ exports.register = (server, options, next) => {
...
+                        .find(n => n.name === nextJobName).id;
+                    const externalBuildId = result.builds.find(b => b.jobId === externalJobId).id;
+                    const externalBuild = await buildFactory.get({ id: externalBuildId });
+                    const masterParentBuilds = {};
+
+                    masterParentBuilds[result.pipelineId] = {
+                        eventId: result.id,
+                        jobs: { [externalJobName]: externalBuildId }
+                    };
+                    const nextJobs1 = workflowParser.getNextJobs(workflowGraph,
```
Comment (jithine) [inline plugins/builds/index.js]: can we have a meaningful name than `nextJobs` ? eg: `nextJobsToTrigger` 


similar feedback for `nextJobName1` & `result1`

File: plugins/builds/index.js [hunk 2]
```diff
@@ -586,6 +596,114 @@ async function handleNewBuild({ done, hasFailure, newBuild }) {
...
+ * @param  {Object}     joinParentBuilds    Parent builds info for join
+ * @param  {Object}     currentJobParentBuilds Parent builds info for current job
+ * @param  {Object}     currentBuildInfo    Parent builds info for current build
+ * @return {Promise}                        The newly updated/created build
+ */
+async function createOrRunNextBuild({ buildFactory, jobFactory, eventFactory, pipelineFactory,
+    pipelineId, jobName, start, username, scmContext, build, event, parentBuilds, parentEventId,
+    externalPipelineId, externalJobName, parentBuildId, isExternal, workflowGraph, nextJobName,
+    externalBuild, joinListNames, joinParentBuilds, currentJobParentBuilds, currentBuildInfo }) {
+    const internalBuildConfig1 = {
```
Comment (jithine) [inline plugins/builds/index.js]: lets not add `1` suffix . same for `externalBuildConfig1`

File: test/plugins/builds.test.js [hunk 3]
```diff
@@ -2264,12 +2264,20 @@ describe('build plugin test', () => {
                     }]);
 
                     return newServer.inject(options).then(() => {
-                        jobCconfig.start = false;
-                        jobCconfig.parentBuilds = { 123: {
-                            jobs: { a: 12345, b: null },
-                            eventId: '8888'
-                        } };
-                        assert.calledWith(buildFactoryMock.create, jobCconfig);
+                        assert.calledWith(buildFactoryMock.create.firstCall, {
```
Comment (jithine) [inline test/plugins/builds.test.js]: does this cover all scenarios ? don't we need more test cases ? Seeing that coverage has decreased.

Comment (tkyi) [reply test/plugins/builds.test.js]: added a test

**Pull request merged**